tests/storage/glusterfs/conftest.py

```diff
@@ -81,6 +82,45 @@ def gluster_disk(host, sr_disk_for_all_hosts):
         h.ssh(['umount', '/mnt/sr_disk'])
         h.ssh(['rmdir', '/mnt/sr_disk'])
 
+def _fallback_gluster_teardown(host):
+    # See: https://microdevsys.com/wp/volume-delete-volume-failed-some-of-the-peers-are-down/
+    # Remove all peers and bricks from the hosts volume and then stop and destroy volume.
+    def teardown_for_host(h):
+        print("< Fallback teardown on host: %s" % h)
+        hosts = h.pool.hosts
+
+        h.ssh(['systemctl', 'restart', 'glusterd'])
+
+        gluster_cmd = ['gluster', '--mode=script', 'volume', 'remove-brick', 'vol0', 'replica', '1']
+        for h2 in hosts:
+            if h.hostname_or_ip != h2.hostname_or_ip:
+                gluster_cmd.append('%s:/mnt/sr_disk/vol0/brick0' % h2.hostname_or_ip)
+        gluster_cmd.append('force')
+        h.ssh(gluster_cmd)
+
+        for h2 in hosts:
+            if h.hostname_or_ip != h2.hostname_or_ip:
+                h.ssh(['gluster', '--mode=script', 'peer', 'detach', h2.hostname_or_ip])
+
+        try:
+            # Volume might already be stopped if failure happened on delete
+            h.ssh(['gluster', '--mode=script', 'volume', 'stop', 'vol0'])
+        except Exception as e:
+            pass
+
+        h.ssh(['gluster', '--mode=script', 'volume', 'delete', 'vol0'])
+
+    try:
+        teardown_for_host(host)
```

Comment:
I got umount errors but that was probably caused by the way I tested (cutting gluster traffic between two hosts), so I think it's OK.